go/mysql/server.go

```diff
@@ -531,6 +545,21 @@ func (l *Listener) handle(conn net.Conn, connectionID uint32, acceptTime time.Ti
 	}
 }
 
+func setTcpConnProperties(conn *net.TCPConn, keepAlivePeriod time.Duration) {
+	if err := conn.SetKeepAlive(true); err != nil {
+		log.Errorf("unable to enable keepalive on tcp connection: %v", err)
+		return
+	}
+
+	if keepAlivePeriod <= 0 {
+		return
+	}
```

Comment:
Shoulnd't this if condition be before we enable `KeepAlives` at all?

Reply:
We should always enable keep-alive on tcp connection.

Reply:
No, I don't agree with that. If the user has explicitly set the flag to 0 or a negative value, it means that they don't want `KeepAlives`. We shouldn't enable it automatically. I think that will be unexpected from the user's point of view.

The default for the flag can be a positive number, so that anyone running Vitess without any modifications to the flag gets `KeepAlives`, but for any user who wants to disable this, they should have the capability of doing that.

Reply:
The reason for keeping this on by default and having no way to disable it.
The reason is that the MySQL server does the same thing and also the SQL clients as well does the same implementation (Ex c client, go client).

I did not find a compelling reason why client would want to turn this off